### PR TITLE
[test] Run rom_e2e_shutdown_output for remaining life cycle states

### LINKED ISSUE #14270: [rom-e2e] rom_e2e_shutdown_output
**Testpoint name:** [rom_e2e_shutdown_output](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson?q=rom_e2e_shutdown_output)

**Contact person:** @alphan

**Description:** Verify that ROM can properly report errors over UART.

- Attempt to boot without a valid ROM_EXT.
- Verify that we can receive the 28 character long (`BFV:xxxxxxxx\r\nLCV:xxxxxxxx\r\n`) error
  output in all life cycle states where Ibex is enabled, i.e. TEST_UNLOCKED*, PROD,
  PROD_END, DEV, and RMA.
    - BFV should be 0142500d for all life cycle states.
    - See the table below for the expected LCV value.

| LC State |     LCV    | OTP LIFE_CYCLE state | OTP LIFE_CYCLE count |
|:--------:|:----------:|:--------------------:|:--------------------:|
|   TEST_UNLOCKED0  | 0x02108421 |   "TEST_UNLOCKED0"   |           5          |
|   DEV    | 0x21084210 |        "DEV"         |           5          |
|   PROD   | 0x2318c631 |        "PROD"        |           5          |
| PROD_END | 0x25294a52 |      "PROD_END"      |           5          |
|   RMA    | 0x2739ce73 |        "RMA"         |           5          |

### PR DESCRIPTION
**Pull request by alphan**
This PR

* Updates the owner OTP partition to report errors without any redaction.
* Moves various constants and the `hex()` function to `const.bzl`
* Runs `rom_e2e_shutdown_output` for remaining life cycle states.

Fixes #14270 

Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

**Review by jrvanwhy — APPROVED** [review]

**Comment by alphan** [comment]
@milesdai @dmcardle can you tell me which tags I need to enable these tests in CI? Thanks!

**Comment by milesdai** [comment]
> can you tell me which tags I need to enable these tests in CI? Thanks!
To echo what we discussed offline, at the moment, you will need to add the `vivado` tag to any splicing tasks and jobs that depend on those splice tags. In your case, add it to `cw310_params` and the test suite. Also add the manual tag to the test suite.

**Comment by milesdai** [comment]
Also, at the moment, CI is currently unable to splice bitstreams on the fly on the FPGA runner, so this PR will probably break during the FPGA tests with an error about `updatemem`. I am working on fixing this in #15891.

**Comment by alphan** [comment]
> Also, at the moment, CI is currently unable to splice bitstreams on the fly on the FPGA runner, so this PR will probably break during the FPGA tests with an error about `updatemem`. I am working on fixing this in #15891.

Thanks for that fix @milesdai! I've just merged that PR. 🤞🏼

File: rules/const.bzl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        BOOT_POLICY = struct(
+            BAD_IDENTIFIER = 0x0142500d,
+            BAD_LENGTH = 0x0242500d,
+            ROLLBACK = 0x0342500d,
+        ),
+        MANIFEST = struct(
+            BAD_ENTRY_POINT = 0x014d410d,
+            BAD_CODE_REGION = 0x024d410d,
+        ),
+    ),
```
Comment (jadephilipoom) [inline rules/const.bzl:48]: These codes are derived from `error.h`, right? If so, it would be good to add a "must match" comment here like the ones above.

**Review by jadephilipoom — APPROVED** [review]
Nice to have a more systematic way of handling these constants!

File: rules/const.bzl (hunk 1 above)
Comment (alphan) [reply rules/const.bzl:48]: Yes, good point, done!

**Comment by alphan** [comment]
image

**Comment by alphan** [comment]
```
//sw/device/silicon_creator/rom/e2e:shutdown_output_dev_fpga_cw310_rom   PASSED in 17.2s
//sw/device/silicon_creator/rom/e2e:shutdown_output_prod_end_fpga_cw310_rom PASSED in 17.3s
//sw/device/silicon_creator/rom/e2e:shutdown_output_prod_fpga_cw310_rom  PASSED in 17.2s
//sw/device/silicon_creator/rom/e2e:shutdown_output_rma_fpga_cw310_rom   PASSED in 17.2s
//sw/device/silicon_creator/rom/e2e:shutdown_output_test_unlocked0_fpga_cw310_rom PASSED in 17.3s
```

**Pull request merged**